src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -304,8 +304,10 @@ private void SolutionEvents_AfterClosing()
             }
         }
 
-        public void CleanCache()
+        public async Task CleanCacheAsync()
         {
+            await SolutionManager.CleanAsync();
```

Comment:
Technically Job does iterate over NuGet projects
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L165

So it's okay for a utility/component to have deeper knowledge of projects.

`SolutionManager` interface should be limited to query methods only. I insist on moving the functionality out. If not worker it can be some other component. Like buildnadler?